diff -Naur ZODB.orig/src/ZODB/blob.py ZODB/src/ZODB/blob.py
--- ZODB/blob.py	2012-06-22 18:38:26.066030225 +0200
+++ ZODB/blob.py	2012-06-22 18:39:48.126016052 +0200
@@ -337,11 +337,11 @@
 
     def create(self):
         if not os.path.exists(self.base_dir):
-            os.makedirs(self.base_dir, 0700)
+            os.makedirs(self.base_dir, 0750)
             log("Blob directory '%s' does not exist. "
                 "Created new directory." % self.base_dir)
         if not os.path.exists(self.temp_dir):
-            os.makedirs(self.temp_dir, 0700)
+            os.makedirs(self.temp_dir, 0750)
             log("Blob temporary directory '%s' does not exist. "
                 "Created new directory." % self.temp_dir)
 
@@ -359,8 +359,8 @@
                     (self.layout_name, self.base_dir, layout))
 
     def isSecure(self, path):
-        """Ensure that (POSIX) path mode bits are 0700."""
-        return (os.stat(path).st_mode & 077) == 0
+        """Ensure that (POSIX) path mode bits are 0750."""
+        return (os.stat(path).st_mode & 027) == 0
 
     def checkSecure(self):
         if not self.isSecure(self.base_dir):
@@ -385,7 +385,7 @@
 
         if create and not os.path.exists(path):
             try:
-                os.makedirs(path, 0700)
+                os.makedirs(path, 0750)
             except OSError:
                 # We might have lost a race.  If so, the directory
                 # must exist now
@@ -891,7 +891,7 @@
             file2.close()
         remove_committed(f1)
     if chmod:
-        os.chmod(f2, stat.S_IREAD)
+        try:
+            os.chmod(f2, stat.S_IRUSR | stat.S_IRGRP)
+        except OSError:
+            pass
 
 if sys.platform == 'win32':
     # On Windows, you can't remove read-only files, so make the
